extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/tls/QuarkusKafkaSslEngineFactory.java

```diff
@@ -0,0 +1,91 @@
+package io.quarkus.kafka.client.tls;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.util.Map;
+import java.util.Set;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.spi.CDI;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.security.auth.SslEngineFactory;
+import org.jboss.logging.Logger;
+
+import io.quarkus.tls.TlsConfiguration;
+import io.quarkus.tls.TlsConfigurationRegistry;
+
+public class QuarkusKafkaSslEngineFactory implements SslEngineFactory {
+
+    private static final Logger log = Logger.getLogger(QuarkusKafkaSslEngineFactory.class);
+
+    private TlsConfiguration configuration;
+    private SSLContext sslContext;
+
+    @Override
+    public SSLEngine createClientSslEngine(String peerHost, int peerPort, String endpointIdentification) {
+        SSLEngine sslEngine = sslContext.createSSLEngine(peerHost, peerPort);
+        sslEngine.setUseClientMode(true);
+        SSLParameters sslParameters = sslEngine.getSSLParameters();
+        sslParameters.setEndpointIdentificationAlgorithm(endpointIdentification);
+        sslEngine.setSSLParameters(sslParameters);
+        return sslEngine;
+    }
+
+    @Override
+    public SSLEngine createServerSslEngine(String peerHost, int peerPort) {
+        throw new IllegalStateException("Server mode is not supported");
+    }
+
+    @Override
+    public boolean shouldBeRebuilt(Map<String, Object> nextConfigs) {
+        return false;
+    }
+
+    @Override
+    public Set<String> reconfigurableConfigs() {
+        return Set.of();
```

Comment:
What is the idea being this method? Is this about certificate renewal ?

Reply:
If I understood correctly those are for dynamically reconfigurable server options. 
For the client, I didn't see any certificate renewal mechanism.